src/App.js

```diff
@@ -1,25 +1,51 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useEffect } from 'react';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import { getRockets } from './redux/rockets/rockets';
+import { getMissions } from './redux/missions/missions';
+
+import Header from './components/Header';
+import Missions from './components/Missions';
+import Rockets from './components/Rockets';
+import Profile from './components/Profile';
+
+const App = () => {
+  const missions = useSelector((state) => state.missions, shallowEqual);
+  const dispatch = useDispatch();
+  const rockets = useSelector((state) => state.rockets, shallowEqual);
+
+  useEffect(() => {
+    dispatch(getRockets());
+    dispatch(getMissions());
```

Comment:
According to the project requirements:

![image](https://user-images.githubusercontent.com/57421823/162109677-667b9d6b-94ea-4d61-b380-56822161158d.png)

Therefore:

- [ ] Please, make sure the "missions" data is only fetched (once) when a user navigates to the Missions section.

## Hints

- Just remove the `dispatch(getMissions());` from this `useEffect`, and create another `useEffect` with the same dispatch in the component that renders the missions.

That way, we will meet this requirement and move closer to approval 💪